### ISSUE TITLE
Revise routes

### ISSUE THREAD

**Issue by gr2m** [hoodiehq/hoodie-server#319]
## tl;dr

we want to replace the current

```
/_api/ ... # CouchDB API
/_api/_files/hoodie.js
/_api/_plugins
/_api/_plugins/pluginname
/_api/_plugins/pluginname/_api/ ... # custom plugin API
```

with 

```
/hoodie # tells version of Hoodie Server
/hoodie/bundle.js # returns generated hoodie.js + plugin code
/hoodie/store/ ... # REST api for hoodie.store
/hoodie/account/ ... # REST api for hoodie.account
/hoodie/pluginname/ ... custom plugin API (no more _api prefix)
```

-----

**original description:**

This is what we currently have

```
/_api/ ... # CouchDB API
/_api/_files/hoodie.js
/_api/_plugins
/_api/_plugins/pluginname
/_api/_plugins/pluginname/_api/ ... # custom plugin API
```

Before 1.0, I'd like to rethink the structure, it will be hard to change afterwards. 

Here's what I'd suggest

```
/hoodie # tells version of Hoodie Server
/hoodie/api.js # returns generated hoodie.js + plugin code
/hoodie/db/ ... # CouchDB API
/hoodie/plugins/
/hoodie/plugin/pluginname
/hoodie/plugin/pluginname/ ... custom plugin API (no more _api prefix)
```

These URLs might not be of much relevance today, but I'm sure they will be more so in future. 

### loading the Hoodie API

Every app will need to load the frontend API with a script tag. I think it's worth to make it clear and simple as possible. And

```html
<script src="https://myapp.example.com/hoodie/api.js"></script>
```

is much cleaner than

```html
<script src="https://myapp.example.com/_api/_files/hoodie.js"></script>
```

We don't really load some kind of library, like you'd load jQuery or underscore. It was started with the library-mind-thinking, therefore it's repository name `hoodie.js`, but it evolved into something different, which is bound directly to the Hoodie Server and its plugins.

In future, its content should be even more dynamic, with plugins providing a UI to set validation rules, which will be automatically added to the frontend API, so a call of `hoodie.store.add` will fail even before creating a new object locally, just as an example.

### The /hoodie prefix

One day, there will be millions of Hoodie apps out there, and thousands of Hoodie Hosters. I think it would be great to see immediately what apps are build with Hoodie, and developers should recognize it immediately in the code bases. `/_api` is generic and could stand for anything, `/hoodie` on the other side is very clear. 

It might be confusing for people starting with Hoodie where this `hoodie` folder is, but the same goes for `/_api`. And the benefits outweights the added explanation it'll take.

**additional thought** We could auto-generate the `www/hoodie` hoodie if it doesn't exist yet. We put in a README.md to explain what it does. It would create `www/hoodie/bundle.js` the first time `/hoodie/bundle.js` is requested – unless it already exists. I guess we do cache the browserified JS file already anyway somewhere, why not make it transparent, and give me a way to put it in my build workflow, so it doesn't take the extra time to generate it in production on first load.

### Plugins

`/hoodie/plugin/users` – get it? It matches the `hoodie-plugin-users` convention of our npm module names. The plugins can expose their own APIs as sub paths, like `/hoodie/plugin/users/confirm?toke=abc4567`, no need for an extra `/_api`

**Comment by boennemann** [comment]
Bumping this because it's very relevant for the 1.0 planning.
